lib/web_ui/lib/src/ui/pointer.dart

```diff
@@ -291,6 +292,84 @@ class PointerData {
         'scrollDeltaY: $scrollDeltaY'
         ')';
   }
+
+  /// serialize the PointerData to a JSON object.
+  Map<String, dynamic> toJson() {
+    // to perform null check on a variable rather than a getter
```

Comment:
A side note: For null-safety, the following is reporting issue for not null-checked
```Dart
class Foo {
  int? value;
  Map<String, int> toJson() {
    return <String, int>{
      if(value != null) 'value': value
    };
  }
}
```
But the following is fine:
```Dart
Map<String, int> foo(int? value) {
  return <String, int>{
    if(value != null) 'value': value
  };
}
```

This is due to `value` in a class is actually getter/setter, which may be overrided, and therefore static analysis cannot determine if it will be null.